src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetVersionMappingControllerTests.cs

```diff
@@ -876,6 +873,469 @@ public async Task DataSetVersionMappingDoesNotExist_Returns404()
+                                OptionMappings =
+                                {
+                                    {
+                                        "Filter 1 option 1 key",
+                                        new FilterOptionMapping
+                                        {
+                                            Source = new MappableFilterOption("Filter 1 option 1 label"),
+                                            Type = MappingType.AutoMapped,
+                                            CandidateKey = "Filter 1 option 1 key"
+                                        }
+                                    },
+                                    {
+                                        "Filter 1 option 2 key",
+                                        new FilterOptionMapping
+                                        {
+                                            Source = new MappableFilterOption("Filter 1 option 2 label"),
+                                            Type = MappingType.ManualNone
+                                        }
+                                    }
+                                }
+                            }
+                        },
+                        {
+                            "Filter 2 key", new FilterMapping
+                            {
+                                Source = new MappableFilter("Filter 2 label"),
+                                Type = MappingType.AutoMapped,
+                                CandidateKey = "Filter 2 key",
+                                OptionMappings =
+                                {
+                                    {
+                                        "Filter 2 option 1 key",
+                                        new FilterOptionMapping
+                                        {
+                                            Source = new MappableFilterOption("Filter 2 option 1 label"),
+                                            Type = MappingType.AutoMapped,
+                                            CandidateKey = "Filter 1 option 1 key"
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersionMappings.Add(mappings);
+            });
+        
+            var client = BuildApp().CreateClient();
+        
+            List<FilterOptionMappingUpdateRequest> updates =
+            [
+                new()
+                {
+                    FilterKey = "Filter 1 key",
+                    SourceKey = "Filter 1 option 1 key",
+                    Type = MappingType.ManualMapped,
+                    CandidateKey = "target-filter-option-1-key"
```

Comment:
I've just realised, we don't have anything in the code which validates that the candidate key selected is a legitimate candidate key. Should we validate it against the list of candidate keys in the mapping plan?